infoschema/tables.go

```diff
@@ -1239,7 +1239,7 @@ func dataForStatisticsInTable(schema *model.DBInfo, table *model.TableInfo) [][]
 				i+1,           // SEQ_IN_INDEX
 				key.Name.O,    // COLUMN_NAME
 				"A",           // COLLATION
-				0,             // CARDINALITY
+				nil,           // CARDINALITY
```

Comment:
Do we need to do this change? @morgo 
This column is `INFORMATION_SCHEMA.STATISTICS.CARDINALITY`. We don't have real data for this column, so I think 0 is OK. Do you have any advice?

Reply:
I prefer `nil`. I didn't realize we could use it in this context, there are some other commands I could fix (show processlist's database name when empty..) :-)

Reply:
> I didn't realize we could use it in this context
I don't quite understand `in this context`. In this PR I get the value of `CARDINALITY` in a test. But I am not sure about the difference between `0` and `nil` in MySQL.

Reply:
OK, I took a look at this in MySQL 8.0.15 and I think I understand why certain ones are `NULL` and others are `0`:

- The performance schema tables are `NULL` cardinality.  This is because the pfs engine does not maintain cardinality stats ever.  So you can think for the classic meaning for `NULL`: unknown, not applicable.

- The other statistics that were `0` were typically empty tables, or tables with out of date statistics.  So zero was the last known good number.

If we don't maintain a true cardinality number, then we are similar to performance schema tables (`NULL`).

Reply:
I got it. Thank you!